test/batchnorm.jl

```diff
@@ -0,0 +1,97 @@
+include("header.jl")
+
+srand(42)
+TOL=0.1
+
+# CPU 4d buffers
+ax4 = Array(5rand(4,5,5,3));
+aw4 = bnparam(5)
+# GPU 4d buffers
+kax4 = KnetArray(5rand(4,5,5,3));
+kaw4 = KnetArray(bnparam(5));
+
```

Comment:
Let's test all element types (Float32,Float64), array types (KnetArray,Array), and dimensions (2,4,5)?